cpp/src/arrow/filesystem/gcsfs.cc

```diff
@@ -688,6 +744,9 @@ Result<std::shared_ptr<io::RandomAccessFile>> GcsFileSystem::OpenInputFile(
 
 Result<std::shared_ptr<io::RandomAccessFile>> GcsFileSystem::OpenInputFile(
     const FileInfo& info) {
+  if (info.IsDirectory()) {
+    return Status::IOError("Cannot open a directory as an input stream");
```

Comment:
same comment above about providing the path?

Reply:
Fixed too.